src/hooks/useFetch.js

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect } from "react";
+
+export const useFetch = (url, method = "GET") => {
+  const [data, setData] = useState(null);
+  const [isPending, setIsPending] = useState(false);
+  const [error, setError] = useState(null);
+  const [options, setOptions] = useState(null);
+
+  const postData = (postData) => {
+    setOptions({
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(postData),
+    });
+  };
+
+  useEffect(() => {
+    const controller = new AbortController();
+
+    const fetchData = async (fetchOptions) => {
+      setIsPending(true);
+
+      try {
+        const res = await fetch(url, {
+          ...fetchOptions,
+          signal: controller.signal,
+        });
+        if (!res.ok) {
+          throw new Error(res.statusText);
+        }
```

Comment:
good that youre catching errors and implementing async js - thats something i still need to learn